### PR TITLE
add blob sidecars store test

### LINKED ISSUE #7946: BlobSidecar related Store tests
Old description of the issue was:

Make a BlobSidecar replacement for tests: `retrieveSignedBlockAndBlobsSidecar_shouldReturnEmptyIfBlockNotPresent`, `retrieveSignedBlockAndBlobsSidecar_withInconsistentSidecar`, `retrieveSignedBlockAndBlobsSidecar_withEmptyBlobs`, `retrieveSignedBlockAndBlobsSidecar_withBlobs` removed in #7085 

Store APIs are now changed but there are some apis not tested in `StoreTest`

### PR DESCRIPTION
**Pull request by mehdi-aouadi**
## PR Description
Adds blob sidecars store tests

## Fixed Issue(s)

fixes #7946 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Review by tbenr — COMMENTED** [review]
I don't think the kzgcommitments-related tests are actually exercising new code paths, since they are part of the block itself. It seems like they tests consistency in our own chain building test fixtures.

Instead, I see `getBlobSidecarsIfAvailable` and `retrieveEarliestBlobSidecarSlot` not covered by any test though...

**Comment by mehdi-aouadi** [comment]
> I don't think the kzgcommitments-related tests are actually exercising new code paths, since they are part of the block itself. It seems like they tests consistency in our own chain building test fixtures.
> 
> Instead, I see `getBlobSidecarsIfAvailable` and `retrieveEarliestBlobSidecarSlot` not covered by any test though...

Should be good now. Added some tests to cover `getBlobSidecarsIfAvailable` and `retrieveEarliestBlobSidecarSlot` (refactored the `ChainUpdated` and the `AbstractStoreTest` in order to call the blobs pruning logic) and refactored the old ones

File: storage/src/test/java/tech/pegasys/teku/storage/store/AbstractStoreTest.java [hunk 1]
```diff
@@ -150,15 +150,52 @@ protected UpdatableStore createGenesisStore() {
...
+  }
+
   protected StoreBuilder createStoreBuilder(final StoreConfig pruningOptions) {
     final SignedBlockAndState genesis = chainBuilder.generateGenesis();
     final Checkpoint genesisCheckpoint = chainBuilder.getCurrentCheckpointForEpoch(0);
+    return createStoreBuilder(pruningOptions, genesis, genesisCheckpoint);
+  }
+
+  private StoreBuilder createStoreBuilder(
+      StoreConfig pruningOptions, SignedBlockAndState genesis, Checkpoint genesisCheckpoint) {
```
Comment (tbenr) [inline storage/src/test/java/tech/pegasys/teku/storage/store/AbstractStoreTest.java]: finals

**Review by tbenr — APPROVED** [review]
LGTM

File: storage/src/test/java/tech/pegasys/teku/storage/store/AbstractStoreTest.java (hunk 1 above)
Comment (mehdi-aouadi) [reply storage/src/test/java/tech/pegasys/teku/storage/store/AbstractStoreTest.java]: My bad! Done

**Pull request merged**